File: python/ex10.py
class Adicionadora:
    def adicionar(self, *valores):
        return sum(valores)


def executar():
    calculadora = Adicionadora()

    resultado_2 = calculadora.adicionar(5, 10)
    print(f"Soma de 5 e 10: {resultado_2}")

    resultado_3 = calculadora.adicionar(5, 10, 15)
    print(f"Soma de 5, 10 e 15: {resultado_3}")


if __name__ == "__main__":
    executar()

File: golang/ex10.go
package main

import (
	"fmt"
)

type Computador struct{}

func (c Computador) SomarDuasValores(x, y float64) float64 {
	return x + y
}

func (c Computador) SomarTresValores(x, y, z float64) float64 {
	return x + y + z
}

func main() {
	computador := Computador{}

	resultado1 := computador.SomarDuasValores(5.5, 3.2)
	fmt.Printf("Resultado da soma de dois valores: %.2f\n", resultado1)

	resultado2 := computador.SomarTresValores(4.0, 2.5, 1.5)
	fmt.Printf("Resultado da soma de três valores: %.2f\n", resultado2)
}
